### PR TITLE
add estimateFee to the JSON RPC 

### PR DESCRIPTION
**Pull request by ArielElp**
This change is [Reviewable](https://reviewable.io/reviews/starkware-libs/starknet-specs/4)

### CODE REVIEW

File: api/starknet_api_openrpc.json [hunk 1]
```diff
@@ -431,6 +431,54 @@
                 }
             ]
         },
+        {
+            "name": "starknet_estimateFee",
+            "summary": "estimate the fee for a given StarkNet transaction",
+            "description": "estimates the resources required by a transaction relative to a given state",
+            "paramStructure": "by-name",
```
Comment (lior-stark) [inline api/starknet_api_openrpc.json:438]: why do you need to specify this?

File: api/starknet_api_openrpc.json [hunk 2]
```diff
@@ -431,6 +431,54 @@
...
+                {
+                    "name": "request",
+                    "summary": "The details of the function call",
+                    "schema": {
+                        "$ref": "#/components/schemas/FUNCTION_CALL"
+                    },
+                    "required": true
+                },
+                {
+                    "name": "block_hash",
```
Comment (lior-stark) [inline api/starknet_api_openrpc.json:449]: so the state assumed is the state at the beginning or end of the block?

**Review by lior-stark — CHANGES_REQUESTED** [review]
This should branched off the 0.10.0 version, which has the getClassXXX methods.
please rebase over master.

**Review by ArielElp — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/starknet-specs/4)* status: 0 of 1 files reviewed, 2 unresolved discussions (waiting on @lior-stark)

___
*[`api/starknet_api_openrpc.json` line 438 at r1](https://reviewable.io/reviews/starkware-libs/starknet-specs/4#-N3No2HvW-TM5BJq2TME-r1-438:-N3NoP6v6FEf_86GOiMo:b26iizn) ([raw file](https://github.com/starkware-libs/starknet-specs/blob/e6275ae34a94aa5c280eb41d5adf533c11588422/api/starknet_api_openrpc.json#L438)):*
<details><summary><i>Previously, lior-stark wrote…</i></summary><blockquote>

why do you need to specify this?
</blockquote></details>

Copied from starknet_call, I have no idea, removed in both locations
___
*[`api/starknet_api_openrpc.json` line 449 at r1](https://reviewable.io/reviews/starkware-libs/starknet-specs/4#-N3No2HvW-TM5BJq2TME-r1-449:-N3NoDOVAtg8MOFS3kDy:b2zwx7l) ([raw file](https://github.com/starkware-libs/starknet-specs/blob/e6275ae34a94aa5c280eb41d5adf533c11588422/api/starknet_api_openrpc.json#L449)):*
<details><summary><i>Previously, lior-stark wrote…</i></summary><blockquote>

so the state assumed is the state at the beginning or end of the block?
</blockquote></details>

End, like in other methods like call.

**Review by lior-stark — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/starknet-specs/4)* status: 0 of 1 files reviewed, all discussions resolved (waiting on @lior-stark)

___
*[`api/starknet_api_openrpc.json` line 449 at r1](https://reviewable.io/reviews/starkware-libs/starknet-specs/4#-N3No2HvW-TM5BJq2TME-r1-449:-N3NyHTS4P6vrdwYh6HG:bivp1g8) ([raw file](https://github.com/starkware-libs/starknet-specs/blob/e6275ae34a94aa5c280eb41d5adf533c11588422/api/starknet_api_openrpc.json#L449)):*
<details><summary><i>Previously, ArielElp wrote…</i></summary><blockquote>

End, like in other methods like call.
</blockquote></details>

I'm not sure we were explicit about this there as well.
we should probably document it somewhere. here?

**Review by ArielElp — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/starknet-specs/4)* status: 0 of 1 files reviewed, all discussions resolved (waiting on @lior-stark)

___
*[`api/starknet_api_openrpc.json` line 449 at r1](https://reviewable.io/reviews/starkware-libs/starknet-specs/4#-N3No2HvW-TM5BJq2TME-r1-449:-N3NzK9A21lrBdYBLKLK:b-7bwp2e) ([raw file](https://github.com/starkware-libs/starknet-specs/blob/e6275ae34a94aa5c280eb41d5adf533c11588422/api/starknet_api_openrpc.json#L449)):*
<details><summary><i>Previously, lior-stark wrote…</i></summary><blockquote>

I'm not sure we were explicit about this there as well.
we should probably document it somewhere. here?
</blockquote></details>

Beginning makes no sense imo, part of the block header is "global_state_root", each block is associated with a state.

**Review by lior-stark — APPROVED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/starkware-libs/starknet-specs/4#-:-N3OwqHU1HN96rlPFK8q:bnfp4nl)* status: 0 of 1 files reviewed, all discussions resolved (waiting on @lior-stark)

**Review by ArielElp — COMMENTED** [review]
> Reviewed 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/starknet-specs/4)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @ArielElp)

**Pull request merged**